### PR TITLE
Add documentation for writing a new AssemblyDocumentation handler

### PR DESCRIPTION
**Pull request by supergrecko**
As an effort to make CE easier to modify, I've added documentation for adding a assembly documentation for a new instruction set.

### CODE REVIEW

File: docs/AddingAssemblyDocumentation.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Adding Assembly Documentation for a new instruction set
+
+This document explains how to add assembly documentation for a new instruction set to Compiler ("CE" from here on).
```
Comment (dkm) [inline docs/AddingAssemblyDocumentation.md:3]:
```suggestion
This document explains how to add assembly documentation for a new instruction set to Compiler Explorer ("CE" from here on).
```

File: docs/AddingAssemblyDocumentation.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+First of all we need to find a data source to get our instruction info from. While it is possible to write down
+information about every single instruction for an instruction set, it's far from maintainable and it is a lot of work.
+
+Existing assembly documentation handlers use some sort of established documentation. The arm32 handler uses the
+developer.arm.com website and the JVM bytecode handler uses Oracle's documentation.
+
+## 2. Create a tool for collecting the data
+
+Since we want to go through the automated route, we should write a script or a piece of code to automatically gather the
+data for us and store it in a nice format that CE expects. The output of the script should be a code generated .js file
```
Comment (dkm) [inline docs/AddingAssemblyDocumentation.md:23]:
```suggestion
data for us and store it in a nice format that CE expects. The output of the script should be a generated .js file
```

**Review by dkm — COMMENTED** [review]
Maybe a very minor style remark would be that there is a mix of "we need to ..." and "you need to....". But that's very clear, you can leave it like this.

Thank you for writing this documentation !

**Comment by supergrecko** [comment]
Great, I've changed some of the we/you wording and included the nits you found.

**Review by dkm — APPROVED** [review]

**Pull request merged**